pyscript.core/src/stdlib/pyscript.py

```diff
@@ -0,0 +1,25 @@
+# export only what we want to expose as `pyscript` module
+# but not what is WORKER/MAIN dependent
+from _pyscript import window, document, IS_WORKER
+
+# this part is needed to disambiguate between MAIN and WORKER
+if IS_WORKER:
+    # in workers the display does not have a default ID
+    # but there is a sync utility from xworker
+    from _pyscript.display import display
+    from _pyscript import sync
+else:
+    # in MAIN both PyWorker and a runtime currentScript.id exist
+    # so these are both exposed and the display, if imported,
+    # will point at the right script as default target
+    PyWorker = window._pyscript.PyWorker
+
+    def __getattr__(attribute_name):
+        if attribute_name == "display":
+            from _pyscript.display import display
+
+            id = window._pyscript.id
+            return lambda *values, target=id, append=True: display(
+                *values, target=target, append=append
+            )
+        raise AttributeError(f"'{__name__}' has no attribute '{attribute_name}'")
```

Comment:
IMO this is a weird thing to do and almost certainly wrong? Maybe I just don't understand the purpose well enough. It only executes once, unless you later delete the name from the module. Module `__getattr__` is perfect for deprecations and lazy initialization, but not appropriate for much else.

Try making a file like the following:
```
_N = 0
def __getattr__(attribute_name):
    global _N
    _N += 1
    return attribute_name + str(_N)
```
and look at what it does... 


Why not:
```py
def display(*values, target=None, append=True):
    if target is None and IS_WORKER:
        raise Error("Need explicit target when using display in worker)
    if target is None:
        target = window._pyscript.id
    element = document.getElementById(target)
    for v in values:
        _write(element, v, append=append)
```
If this doesn't bind `target` correctly then maybe do something in `onBeforeRun` or something...

Reply:
> IMO this is a weird thing to do and almost certainly wrong? ... It only executes once

OK, this might explain @antocuni use case ... let's say currently I was focused on *one* script in main and no more than one (but I will test in a follow up MR more than one script) but the fact a getter is executed only once surprises me so that indeed that's not what we want there.


> Maybe I just don't understand the purpose well enough.

Before the *JS module* had a getter so that if you would `from pyscript import display` that reference would've been the getter and executed every single time and remain the right reference with the right default target to invoke.

Here we're now using a "pure Python module" and if I understand you correctly there's no way in Python to have a getter as module entry.

This leaves me with no options to bring in a default target unless I completely re-think, re-write, or revisit, the current `display` main logic but it'd be weird to make it *JS* dependent in Python, something I was trying to avoid doing with the current branch.

@ntoll or any other, is there any alternative way to have an always different runtime returned `display` out of a module?

To provide more context, I could somehow provide the current ID via JS while the callback executes but then we have async code or async methods that might execute at any time and there's no way to share as pin-pointed default target ID any value when those async calls happen ... reason I was hoping to have instead a getter to trap display once.